### PR TITLE
Users/edzambra/s157/backlogcontextmenu

### PR DESCRIPTION
**Pull request by ezambra**

### CODE REVIEW

File: src/PortfolioPlanning/Common/Services/PortfolioPlanningDataService.ts [hunk 1]
```diff
@@ -399,6 +401,255 @@ export class PortfolioPlanningDataService {
...
+                    props
+                );
+
+                return null;
+            }
+
+            const allProjectConfigPromises: Promise<IProjectConfiguration>[] = [];
+            const newProjectConfigsByProjectId: { [projectIdKey: string]: IProjectConfiguration } = {};
+
+            //  Index by project.
```
Comment (ezambra) [inline src/PortfolioPlanning/Common/Services/PortfolioPlanningDataService.ts]: work item id

File: src/PortfolioPlanning/AddToPlanAction.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        if (context.ids) {
+            result = context.ids;
+            return result;
+        }
+
+        if (context.id) {
+            result.push(context.id);
+            return result;
+        }
```
Comment (shutchings) [inline src/PortfolioPlanning/AddToPlanAction.tsx]: Wow sooo many options on context haha.

File: src/PortfolioPlanning/Common/Services/PortfolioPlanningDataService.ts [hunk 3]
```diff
@@ -399,6 +401,255 @@ export class PortfolioPlanningDataService {
...
+                    const projectIdKey = workItemInfo.ProjectSK.toLowerCase();
+                    const workItemTypeKey = workItemInfo.WorkItemType.toLowerCase();
+
+                    if (plan.projects[projectIdKey]) {
+                        if (!plan.projects[projectIdKey].Items) {
+                            plan.projects[projectIdKey].Items = {};
+                        }
+
+                        if (!plan.projects[projectIdKey].Items[newWorkItemId]) {
+                            //  Check if we have work item type's data.
```
Comment (ezambra) [inline src/PortfolioPlanning/Common/Services/PortfolioPlanningDataService.ts]: remove comment

File: src/PortfolioPlanning/AddToPlanAction.tsx (hunk 2 above)
Comment (ezambra) [reply src/PortfolioPlanning/AddToPlanAction.tsx]: Consistency in our product :-)

**Pull request merged**